sr_port/iorm_wtone.c

```diff
@@ -38,19 +39,13 @@ void iorm_wtone(int ch)
 		switch (io_curr_device.out->ochset)
 		{
 			case CHSET_UTF8:
-				endptr = UTF8_WCTOMB(ch, uni_buf);
-				break;
 			case CHSET_UTF16:
-				/* iorm_write will write BE BOM if first write */
-				/* continue as if UTF16BE */
 			case CHSET_UTF16BE:
-				endptr = UTF16BE_WCTOMB(ch, uni_buf);
-				break;
 			case CHSET_UTF16LE:
-				endptr = UTF16LE_WCTOMB(ch, uni_buf);
+				endptr = UTF8_WCTOMB(ch, uni_buf);
```

Comment:
Something to do with http://tinco.pair.com/bhaskar/gtm/doc/articles/GTM_V6.3-004_Release_Notes.html#GTM-8923 ?

Reply:
Yes it is related to GTM-8923. Previously, we were doing UTF16LE_WCTOMB/UTF16BE_WCTOMB etc. once in iorm_wtone.c and again in iorm_write.c but before that a UTF8_MBTOWC was done in iorm_write and if the conversion did not happen fine then a BADCHAR was incorrectly issued (and the conversion would not happen fine because we are trying to treat a UTF16 byte sequence from iorm_wtone.c as a UTF8 byte sequence in iorm_write.c). That has been fixed by avoiding the UTF16 conversion of the input codepoint "ch" in iorm_wtone.c but converting it to a UTF8 multi-byte sequence and then converting it back to a codepoint in iorm_write.c and then doing the UTF16LE_WCTOMB there. Definitely a lot of unnecessary conversions in the UTF16 case. But those are performance issues, not correctness ones.